### DISCUSSION TITLE
How to deploy ADF with Self-hosted Integration Runtimes?

### DISCUSSION THREAD

**Discussion by NowinskiK** [Azure-Player/azure.datafactory.tools#101] [Q&A] [answered]
[Klinok](http://www.youtube.com/channel/UCaspkOq4wcevALU98u0AfDg) asked:
What about Self-hosted Integration Runtimes?
IR does not work in the new Data Factory after deploying.
What is your idea how to fix it?

This question comes from [my YouTube channel](https://www.youtube.com/watch?v=F8kF9h86_XE&lc=UgyrODJpfVi6KAlkt2Z4AaABAg).

**Comment by NowinskiK** [comment, marked as answer]
Self-Hosted Integration Runtime is a specific scenario. 
There are few steps you have to do in appropriate order:
1) Create empty ADF 
2) Deploy SHIR on your target machine (which will add SHIR to your ADF)
3) Deploy whole ADF without SHIR object

I will prepare a blog post or/and video about that soon.
Vote for this topic to increasing my motivation ;)

**Reply by colin-sqlz** [reply]
how do you do the deploy without SHIR object ?

**Reply by NowinskiK** [reply]
Juts exclude that one object from deployment, using file or textbox, for instance:
https://github.com/SQLPlayer/azure.datafactory.tools#includes--excludes-rules-in-a-file
